### PR TITLE
cookie now respects spaces in comment

### PR DESCRIPTION
**Pull request by koebi**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -99,8 +119,23 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
...
+		Kommentar string
+	}{
+		nick,
+		kommentar,
+	}
+
+	cookie, err := json.Marshal(keks)
+	if err != nil {
+		log.Printf("Could not marshal: %v\n", err)
+		writeError(400, res, "Error: %v", err)
```
Comment (Merovius) [inline main.go]: needs a return

File: main.go [hunk 2]
```diff
@@ -60,12 +62,30 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
+		cookie := struct {
+			Nick      string
+			Kommentar string
+		}{
+			nick,
+			kommentar,
+		}
+		if err = json.Unmarshal(keks, cookie); err != nil {
+			log.Println(err)
+			return err
```
Comment (Merovius) [inline main.go]: That shouldn't even compile

File: main.go [hunk 3]
```diff
@@ -60,12 +62,30 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
-		z.Nick = cookie.Value
-	}
-	z = GetZusage(z.Nick)
-	if cookie, _ := req.Cookie("kommentar"); cookie != nil {
-		z.Kommentar = cookie.Value
+
+	if c, _ := req.Cookie("yarpnarp"); c != nil {
+		keks, err := base64.StdEncoding.DecodeString(c.Value)
+		if err != nil {
+			log.Println(err)
```
Comment (Merovius) [inline main.go:74]: You need to write _something_ to the http-connection (i.e. do http.Error(…) here)

Comment (Merovius) [inline main.go:74]: writeError

File: main.go [hunk 4]
```diff
@@ -60,12 +62,30 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
+		}{
+			nick,
+			kommentar,
+		}
+		if err = json.Unmarshal(keks, cookie); err != nil {
+			log.Println(err)
+			return err
+		}
+		z.Nick = cookie.Nick
+		z = GetZusage(z.Nick)
```
Comment (Merovius) [inline main.go]: Why do we even load this from the database? Why not put it in the cookie?

File: main.go [hunk 5]
```diff
@@ -60,12 +62,30 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
-		z.Kommentar = cookie.Value
+
+	if c, _ := req.Cookie("yarpnarp"); c != nil {
+		keks, err := base64.StdEncoding.DecodeString(c.Value)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		cookie := struct {
```
Comment (Merovius) [inline main.go]: I would define a global Cookie-type, so that you don't have to change it in two places at once.

File: main.go (hunk 2 above)
Comment (koebi) [inline main.go]: Done

File: main.go [hunk 6]
```diff
@@ -99,8 +119,23 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
...
-	http.SetCookie(res, &http.Cookie{Name: "kommentar", Value: kommentar, Expires: time.Date(2030, 0, 0, 0, 0, 0, 0, time.Local)})
+	keks := struct {
+		Nick      string
+		Kommentar string
+	}{
+		nick,
+		kommentar,
+	}
+
+	cookie, err := json.Marshal(keks)
```
Comment (Merovius) [inline main.go]: With a global cookie-type, you can do this in-place and get rid of keks

File: main.go (hunk 1 above)
Comment (koebi) [inline main.go]: done

**Comment by Merovius** [comment]
Please do a rewrite, change the commit-message and force-push. \n is not appropriate to have.

**Comment by Merovius** [comment]
(git rebase -i HEAD~3 or something like that and edit the respective commit)

File: main.go (hunk 5 above)
Comment (koebi) [inline main.go]: done

File: main.go (hunk 6 above)
Comment (koebi) [inline main.go]: done

File: main.go (hunk 3 above)
Comment (koebi) [inline main.go:74]: done

File: main.go [hunk 7]
```diff
@@ -60,12 +67,26 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
-	}
-	z = GetZusage(z.Nick)
-	if cookie, _ := req.Cookie("kommentar"); cookie != nil {
-		z.Kommentar = cookie.Value
+
+	if c, _ := req.Cookie("yarpnarp"); c != nil {
+		keks, err := base64.StdEncoding.DecodeString(c.Value)
+		if err != nil {
+			log.Println(err)
+			writeError(500, res, "Something went wrong: %v", err)
```
Comment (Merovius) [inline main.go]: Don't write err to the client, just log it.

File: main.go [hunk 8]
```diff
@@ -60,12 +67,26 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
+			return
+		}
+
+		cookie := Cookie{}
+
+		if err = json.Unmarshal(keks, cookie); err != nil {
+			log.Println(err)
```
Comment (Merovius) [inline main.go:82]: Like above: writeError

File: main.go [hunk 9]
```diff
@@ -60,12 +67,26 @@ func YarpNarpHandler(res http.ResponseWriter, req *http.Request) {
...
+		keks, err := base64.StdEncoding.DecodeString(c.Value)
+		if err != nil {
+			log.Println(err)
+			writeError(500, res, "Something went wrong: %v", err)
+			return
+		}
+
+		cookie := Cookie{}
+
+		if err = json.Unmarshal(keks, cookie); err != nil {
```
Comment (Merovius) [inline main.go]: You need to pass cookie via pointer

File: main.go [hunk 10]
```diff
@@ -99,8 +120,16 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
 
 	RunHook()
 
-	http.SetCookie(res, &http.Cookie{Name: "nick", Value: nick, Expires: time.Date(2030, 0, 0, 0, 0, 0, 0, time.Local)})
-	http.SetCookie(res, &http.Cookie{Name: "kommentar", Value: kommentar, Expires: time.Date(2030, 0, 0, 0, 0, 0, 0, time.Local)})
+	cookie, err := json.Marshal(Cookie{Nick: nick, Kommentar: kommentar})
+	if err != nil {
+		log.Printf("Could not marshal: %v\n", err)
+		writeError(400, res, "Error: %v", err)
```
Comment (Merovius) [inline main.go]: like above, don't write error to client, just log it

File: main.go (hunk 9 above)
Comment (koebi) [inline main.go]: done

File: main.go (hunk 7 above)
Comment (koebi) [inline main.go]: done

File: main.go (hunk 10 above)
Comment (koebi) [inline main.go]: done

**Pull request merged**